konflux-files/pyproject.toml

```diff
@@ -0,0 +1,18 @@
+[tool.poetry]
```

Comment:
I don't think we need poetry here. IMO, this is adding an unnecessary tool.

AFAIK, the `requirements.txt` file is pip's native lockfile. You do need to install the deps locally to generate it. Thus, you'd need to run the `pip install` [commands from the dockerfile](https://github.com/opendatahub-io/modelmesh-runtime-adapter/pull/65/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L164-L169) locally (preferably in an empty virtualenv). Then, you'd generate the file with `pip freeze > requirements.txt`  which you should commit to the repository.

Then, the Dockerfile needs to be changed to run `pip install -r requirements.txt` instead of the `pip install {library}` commands. The resulting Dockerfile and PR should be compatible with konflux, and it also should be suitable for upstream.

There is a small guide here for the requirements.txt file: https://learnpython.com/blog/python-requirements-file/.

Or, maybe, you are seeing a benefit from using poetry that I'm not seeing?

Reply:
`pip freeze` generates it wrongly, no?

we need something like this:
```
absl-py==2.1.0 ; python_version >= "3.8" and python_version < "3.12" \
    --hash=sha256:526a04eadab8b4ee719ce68f204172ead1027549089702d99b9059f129ff1308 \
    --hash=sha256:7820790efbb316739cde8b4e19357243fc3608a152024288513dd968d7d959ff
astunparse==1.6.3 ; python_version >= "3.8" and python_version < "3.12" \
```

pip freeze generates:
```
archspec @ file:///croot/archspec_1697725767277/work
asgi-logger==0.1.0
asgiref==3.8.1
attrs==23.2.0
boltons @ file:///Users/builder/cbouss/perseverance-python-buildout/croot/boltons_1699240838368/work

```

And the packages needs all to be installed before hand, otherwise, you'll get this:

```
$ pip freeze -r requirements.txt> another.txt
WARNING: Requirement file [requirements.txt] contains grpcio==1.65.5, but package 'grpcio' is not installed
WARNING: Requirement file [requirements.txt] contains h5py==3.10.0, but package 'h5py' is not installed
WARNING: Requirement file [requirements.txt] contains tensorflow==2.13.1, but package 'tensorflow' is not installed
WARNING: Requirement file [requirements.txt] contains docutils==0.20.1, but package 'docutils' is not installed
WARNING: Requirement file [requirements.txt] contains tensorflow-io-gcs-filesystem==0.34.0, but package 'tensorflow-io-gcs-filesystem' is not installed
```


I didn't find any other easier way than rely on poetry to do that.

Reply:
even for pip, you need to install `pip-compile`, another tool. :)
and, it requires everything to be pre-installed before generating the hashes.

Otherwise, a error like this will happen:
```
pip._internal.exceptions.DistributionNotFound: No matching distribution found for tensorflow==2.13.1
```

If we want to automate this task at some point, poetry will be the best is it resolves all dependencies without the need of installing them before.

Reply:
> pip freeze generates it wrongly, no?

Are you sure that we need the hashes. A minimal requriements file seems to be normal. E.g.: https://github.com/opendatahub-io/vllm/blob/main/requirements-common.txt

Reply:
looks like there is a workaround :D
will get back to this on Monday

Reply:
yeah, we don't need poetry.